### PR TITLE
Enhance definitions in TypeScript component

### PR DESCRIPTION
**Pull request by 19majkel94**

### CODE REVIEW

**Comment by RunDevelopment** [comment]
Thank you for this PR!

But you seem to have forgotten to add the minified files. Could you please commit them as well?

**Comment by 19majkel94** [comment]
Sorry, I've used GitHub editor to edit the file in browser.
Will find contribution guidelines and fix this PR soon 😉

**Comment by RunDevelopment** [comment]
If you want to complete the list of builtins and keywords you could also add the `is` and `keyof` keyword as well as the `unknown` and `never` builtin.

**Comment by 19majkel94** [comment]
That makes sense, I've only checked my docusaurus documentation for missing stuffs.
I've pushed the suggested changes 😉 

I miss the most the special keyword colouring feature (violet keywords on screenshot below) but I guess it would require changes in core to support this 😞 

code_2018-08-14_21-14-54

**Comment by RunDevelopment** [comment]
I also think that the syntax coloring of VS Code is just gorgeous.

But you don't have to change the core for that. All you need is a little hook like this:
```js
Prism.hooks.add('after-tokenize', function (env) {
	if (env.language !== 'typescript') {
		return;
	}

	for (var i = 0; i < env.tokens.length; i++) {
		var token = env.tokens[i];

		if (typeof token === 'string')
			continue;

		if (token.type === 'keyword' && /^(?:return|if|else|other special keywords)$/.test(token.content)) {
			var specialKeywordAlias = 'specialKeyword';

			if (!token.alias)
				token.alias = [specialKeywordAlias];
			else if (typeof token.alias === 'string')
				token.alias = [token.alias, specialKeywordAlias];
			else
				token.alias.push(specialKeywordAlias);
		}
	}
});
```

Add this hook before you do the highlighting and all keywords in matching the regular expression will get an additional CSS class which can be used to change the color of certain keywords.

**Comment by 19majkel94** [comment]
@RunDevelopment Maybe I will make a separate PR with this feature in the future 😉

**Comment by RunDevelopment** [comment]
@19majkel94 Maybe you should open an issue for that first.

This feature will not only change the feel of Prism (if we were to also change the themes) but it will also be quite hard to maintain with my quick and dirty solution.

A better way to achieve this would be to split the keyword lists into two and to mark special keywords with an alias. If we were to do this (and also were to change the themes), we should probably add this for a bunch of languages to keep the feel of Prism consistent. 

I think? 
Others might have better ideas (that don't require to change like 50 languages...).

**Comment by mAAdhaTTah** [comment]
Can we get tests for the new keywords?

**Comment by 19majkel94** [comment]
Can we get contribution guide with the info about how to write tests for the new keywords?

**Comment by mAAdhaTTah** [comment]
Captured in #1533. You should be able to check out the other `.feature` files and see how the top part (where the code goes) translate to the bottom part (the list of tokens). Keywords should be fairly straightforward, but let me know if you have issues.

**Comment by 19majkel94** [comment]
@mAAdhaTTah The tests has been updated 🎉

File: tests/languages/typescript/keyword_feature.test [hunk 1]
```diff
@@ -22,42 +29,52 @@ import
...
 typeof
 var
 void
 while
 with
 yield
-module
-declare
-constructor
-enum
```
[the remark this answers is not in the document]
Comment (19majkel94) [reply tests/languages/typescript/keyword_feature.test]: Not removed, just sorted alphabetically. I've added the missing cases too.

[the remark this answers is not in the document]
Comment (mAAdhaTTah) [reply tests/languages/typescript/keyword_feature.test]: Ah I see. `enum` appears twice on the list, so I didn't see it added.

**Comment by mAAdhaTTah** [comment]
Thank you for the contribution!

**Pull request merged**